### ISSUE TITLE
Make sure the framework works with Mac dmg Installer

### ISSUE THREAD

**Issue by bachmanm** [graphaware/neo4j-framework#41]
it does not seem to

**Comment by b3n01t** [comment]
No, it really does not seem to.

**Comment by bachmanm** [comment]
you know the workaround right? Just use the .tgz distro... we'll look into it soon

**Comment by MerrickClark** [comment]
This would be handy info in the Readme!

**Comment by SCKelemen** [comment]
So, did you guys look into it....?

**Comment by FylmTM** [comment]
Hi @bachmanm ,

### TL;DR;
1) Graphaware Framework works with DMG installer
2) Put your stuff into `/Applications/Neo4j\ Community\ Edition.app/Contents/Resources/app/plugins/` directory.

### Detailed information

Configuration window shows that plugin should be placed into `/Users/<user>/Documents/Neo4j/default.graphdb/plugins` directory.

> _Configuration **lies**_.

Plugins are loaded [only from `plugin` directory in application distribution](https://github.com/neo4j/neo4j/blob/3.1/packaging/neo4j-desktop/neo4j-desktop.install4j#L119).

We can check `Info.plist` to verify that:
```
# less /Applications/Neo4j\ Community\ Edition.app/Contents/Info.plist
<key>ClassPath</key>
<string>$APP_PACKAGE/Contents/Resources/app/.install4j/i4jruntime.jar:$APP_PACKAGE/Contents/Resources/app/bin/neo4j-desktop-3.0.4.jar:$APP_PACKAGE/Contents/Resources/app/plugins/*</string><!-- I4J_INSERT_CLASSPATH -->
```

**Why configuration lies?**

Well. To be honest - configuration does not lie. It expects that script that starts Neo4j will respect configuration for plugin directory. We can [check that Neo4j is doing that in standalone mode](https://github.com/neo4j/neo4j/blob/3.1/packaging/standalone/src/main/distribution/shell-scripts/bin/neo4j-shared.sh#L165). 

DMG installer is just hardcoded do specific location and is not doing runtime resolution, like it's big standalone brother.

Why configuration is pointing somewhere else?
- [How plugin directory setting is used](https://github.com/neo4j/neo4j/blob/3.1/packaging/neo4j-desktop/src/main/java/org/neo4j/desktop/model/DesktopModel.java#L196)
- [How plugin directory settings is declared](https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java#L212)
- [How plugin directory setting is resolved](https://github.com/neo4j/neo4j/blob/3.1/community/kernel/src/main/java/org/neo4j/kernel/configuration/Settings.java#L240)

So, Neo4j is optimistically assuming that plugins are somewhere in `neo4j_home`.

I assume that same applies to Windows `.exe` installer and you put plugin somewhere into `Progra Files`.

**Comment by ikwattro** [comment]
Thanks @FylmTM , I think we can add a temporary note in our documentation.

NB: Nice sunglasses ;-)

**Comment by bachmanm** [comment]
Thanks a lot @FylmTM !!

**Comment by SCKelemen** [comment]
This worked fine for me.

**Comment by bachmanm** [comment]
Hey @aldrinm - could you please try this on Windows? :)

**Comment by ikwattro** [comment]
@FylmTM Is there also something specific regarding the conf files? for eg sth related to this https://github.com/graphaware/neo4j-to-elasticsearch/issues/43#issuecomment-252421828

**Comment by FylmTM** [comment]
Heya @ikwattro ,

Actually, I don't remember. There is something with configuration files for sure.
Some configuration are resolved by `neo4j` script in normal distribution.
Other configuration options are parsed on server bootstrap.

**Comment by bachmanm** [comment]
documented thanks to @FylmTM
